models/agent_helpers.go

```diff
@@ -342,6 +342,38 @@ func ChangeAgent(q *reform.Querier, agentID string, params *ChangeCommonAgentPar
 	return row, nil
 }
 
+// PMMAgentsForNode gets pmm-agent for node.
+func PMMAgentsForNode(q *reform.Querier, nodeID string) ([]*Agent, error) {
+	return findPmmAgentsForNode(q, nodeID)
+}
+
+// PMMAgentsForService gets pmm-agent for service.
+func PMMAgentsForService(q *reform.Querier, serviceID string) ([]*Agent, error) {
+	serviceRecord, err := q.SelectOneFrom(ServiceTable, "WHERE service_id = $1", serviceID)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to select Service")
+	}
+	s := serviceRecord.(*Service)
+
+	return findPmmAgentsForNode(q, s.NodeID)
```

Comment:
Do you mean, `pmm-agent` can be run for another node? Here I try to find exactly `pmm-agent` for this node. Also, I don't know how to find `pmm-agent` in another way, as the only thing they are connected is `runs_on_node_id` field. Isn't it?

Reply:
We can find pmm-agent for service using agents. This way :)
Service -> AgentService -> Agent -> Pmm-Agent

For example:
We have 2 nodes: Node 1 and Node 2
Service runs on Node 1, and service has 2 agents.
That 2 agents run on pmm-agent which runs on Node 2.

Reply:
Did you mean this? Oh gosh. It's insane, isn't it?

![IMG_4723](https://user-images.githubusercontent.com/1532188/57699532-9b8a9780-7660-11e9-9708-df9938cb55d2.jpg)

Reply:
It might look insane now, but as soon as we add remote and RDS nodes it will be the only way to monitor them.
So we can rename Node 1 as Remote RDS Node and it will make more sense :)

Reply:
I understand. It makes sense, but It seems so awful to find pmm_agent via children. It makes sense but something wrong here... 😃 Ok, I'll try to go this way...

Reply:
@AlekSi What should I do here?